astropy/wcs/wcs.py

```diff
@@ -129,6 +130,8 @@
 # Additional relax bit flags
 WCSHDO_SIP = 0x10000
 
+# Regular expression defining SIP keyword
+SIP_KW = re.compile('''^(A|B|AP|BP)_(?:[0-9]|1[0-9])_(?:[0-9]|1[0-9])(|[A-Z])$''')
```

Comment:
I believe the trailing letter is there to cover the case of an alternate WCS.
My interpretation of the alternate WCS is that it should not refer to distortion, i.e. a header may have several alternate WCS representations but they share the same distortion. But I've heard the opposite opinions. And I don't think the standard covers what should be done in this case.

That said, the goal here is to make sure no coefficients are passed to wcslib. So even if the header has SIP coefficients with the tailing letter for alternate WCS, I think it's still OK to remove all of them.

wcslib (currently) does not look at `A_ORDER`, ... keywords. I'm guessing that's why they are not removed but I'd be OK with removing them as well.

Is 19 the limit of the polynomial order? Would limiting it to19 instead of just removing all have any impact on performance? If not then perhaps it's OK to keep it like this.  That limit should perhaps be checked for in `_read_sip_kw`. Or perhaps not, since it's somewhat arbitrary. As it stands `astropy.wcs` will work with any polynomial order.